diff --git a/node_modules/react-native-reanimated/react-native-reanimated.d.ts b/node_modules/react-native-reanimated/react-native-reanimated.d.ts
index 29561c5..40e8a36 100644
--- a/node_modules/react-native-reanimated/react-native-reanimated.d.ts
+++ b/node_modules/react-native-reanimated/react-native-reanimated.d.ts
@@ -20,6 +20,7 @@ declare module 'react-native-reanimated' {
     ScrollView as ReactNativeScrollView,
     NativeScrollEvent,
     NativeSyntheticEvent,
+    ColorValue
   } from 'react-native';
   import {
     GestureHandlerGestureEvent,
@@ -199,13 +200,13 @@ declare module 'react-native-reanimated' {
             ? ReadonlyArray<AnimateStyle<S[K][0]>>
             : S[K] extends object
             ? AnimateStyle<S[K]> // allow `number` where `string` normally is to support colors
-            : S[K] extends (string | undefined)
+            : S[K] extends (ColorValue | undefined)
             ? S[K] | number
             :
                 | S[K]
                 | AnimatedNode<
                     // allow `number` where `string` normally is to support colors
-                    S[K] extends (string | undefined) ? S[K] | number : S[K]
+                    S[K] extends (ColorValue | undefined) ? S[K] | number : S[K]
                   >);
     };
 
