diff --git a/node_modules/react-native-reanimated/react-native-reanimated.d.ts b/node_modules/react-native-reanimated/react-native-reanimated.d.ts
index 29561c5..af4ac6f 100644
--- a/node_modules/react-native-reanimated/react-native-reanimated.d.ts
+++ b/node_modules/react-native-reanimated/react-native-reanimated.d.ts
@@ -2,7 +2,7 @@
 // TypeScript Version: 2.8
 
 declare module 'react-native-reanimated' {
-  import { ComponentClass, ReactNode, Component, RefObject } from 'react';
+  import { ComponentClass, ReactNode, Component, RefObject, ComponentType, ComponentProps } from 'react';
   import {
     ViewProps,
     TextProps,
@@ -20,6 +20,8 @@ declare module 'react-native-reanimated' {
     ScrollView as ReactNativeScrollView,
     NativeScrollEvent,
     NativeSyntheticEvent,
+    ColorValue,
+    OpaqueColorValue
   } from 'react-native';
   import {
     GestureHandlerGestureEvent,
@@ -52,10 +54,12 @@ declare module 'react-native-reanimated' {
       IDENTITY = 'identity',
     }
 
+    type ExtrapolateParameter = Extrapolate | { extrapolateLeft?: Extrapolate, extrapolateRight?: Extrapolate }
+
     interface InterpolationConfig {
       inputRange: ReadonlyArray<Adaptable<number>>;
       outputRange: ReadonlyArray<Adaptable<number | string>>;
-      extrapolate?: Extrapolate;
+      extrapolate?: ExtrapolateParameter;
       extrapolateLeft?: Extrapolate;
       extrapolateRight?: Extrapolate;
     }
@@ -68,12 +72,8 @@ declare module 'react-native-reanimated' {
       interpolate(config: InterpolationConfig): AnimatedNode<number>;
     }
 
-    type RawSharedValue = number | string | boolean | object;
-    type SharedValueType = RawSharedValue | RawSharedValue[];
-    export type SharedValue<T extends SharedValueType> = {
-      value: T;
-    };
-
+    export type SharedValue<T> = { value: T };
+    export type DerivedValue<T> = Readonly<SharedValue<T>>;
     export type Mapping = { [key: string]: Mapping } | Adaptable<any>;
     export type Adaptable<T> =
       | T
@@ -192,7 +192,7 @@ declare module 'react-native-reanimated' {
     };
     export type AnimatedTransform = (AdaptTransforms<TransformStyleTypes>)[];
 
-    export type AnimateStyle<S extends object> = {
+    export type AnimateStyle<S> = {
       [K in keyof S]: K extends 'transform'
         ? AnimatedTransform
         : (S[K] extends ReadonlyArray<any>
@@ -210,15 +210,12 @@ declare module 'react-native-reanimated' {
     };
 
     export type AnimateProps<
-      S extends object,
-      P extends {
-        style?: StyleProp<S>;
-      }
+      P extends object
     > = {
       [K in keyof P]: K extends 'style'
-        ? StyleProp<AnimateStyle<S>>
+        ? StyleProp<AnimateStyle<P[K]>>
         : P[K] | AnimatedNode<P[K]>;
-    };
+    } & { animatedProps?: AnimateProps<P> };
 
     type CodeProps = {
       exec?: AnimatedNode<number>;
@@ -227,22 +224,24 @@ declare module 'react-native-reanimated' {
     };
 
     // components
-    export class View extends Component<AnimateProps<ViewStyle, ViewProps>> {
+    export class View extends Component<AnimateProps<ViewProps>> {
       getNode(): ReactNativeView;
     }
-    export class Text extends Component<AnimateProps<TextStyle, TextProps>> {
+    export class Text extends Component<AnimateProps<TextProps>> {
       getNode(): ReactNativeText;
     }
-    export class Image extends Component<AnimateProps<ImageStyle, ImageProps>> {
+    export class Image extends Component<AnimateProps<ImageProps>> {
       getNode(): ReactNativeImage;
     }
     export class ScrollView extends Component<
-      AnimateProps<ViewStyle, ScrollViewProps>
+      AnimateProps<ScrollViewProps>
     > {
       getNode(): ReactNativeScrollView;
     }
     export class Code extends Component<CodeProps> {}
-    export function createAnimatedComponent(component: any): any;
+    export function createAnimatedComponent<P>(
+      component: ComponentClass<P>
+    ): ComponentType<AnimateProps<P>>;
   
     // classes
     export {
@@ -405,12 +404,12 @@ declare module 'react-native-reanimated' {
       easing?: EasingFunction;
     }
     export function withTiming(
-      toValue: number,
+      toValue: number | Exclude<ColorValue, OpaqueColorValue>,  // string as a color value like `"rgba(20,20,20,0)"`
       userConfig?: WithTimingConfig,
       callback?: (isFinished: boolean) => void
     ): number;
     export function withSpring(
-      toValue: number,
+      toValue: number | Exclude<ColorValue, OpaqueColorValue>, // string as a color value like `"rgba(20,20,20,0)"`
       userConfig?: WithSpringConfig,
       callback?: (isFinished: boolean) => void
     ): number;
@@ -418,14 +417,15 @@ declare module 'react-native-reanimated' {
       userConfig: WithDecayConfig,
       callback?: (isFinished: boolean) => void
     ): number;
-    export function cancelAnimation<T extends SharedValue<SharedValueType>>(
-      sharedValue: T
+    export function cancelAnimation<T>(
+      sharedValue: SharedValue<T>
     ): void;
     export function withDelay(delayMS: number, delayedAnimation: number): number;
     export function withRepeat(
       animation: number,
       numberOfReps?: number,
-      reverse?: boolean
+      reverse?: boolean,
+      callback?: (isFinished: boolean) => void
     ): number;
     export function withSequence(...animations: [number, ...number[]]): number;
 
@@ -447,6 +447,13 @@ declare module 'react-native-reanimated' {
     export function runOnJS<A extends any[], R>(
       fn: (...args: A) => R
     ): (...args: Parameters<typeof fn>) => void;
+    
+    type PropsAdapterFunction = (props: Record<string, unknown>) => void;
+    export function createAnimatedPropAdapter(
+      adapter: PropsAdapterFunction,
+      nativeProps?: string[]
+    ): PropsAdapterFunction;
+
     export function processColor(color: number | string): number;
     export function createWorklet<A extends any[], R>(
       fn: (...args: A) => R
@@ -454,12 +461,12 @@ declare module 'react-native-reanimated' {
 
     export function interpolateColor(
       value: number,
-      inputRange: number[],
-      outputRange: string[],
+      inputRange: readonly number[],
+      outputRange: readonly (string | number)[],
       colorSpace?: 'RGB' | 'HSV'
     ): string | number;
 
-    export function makeMutable<T extends SharedValueType>(
+    export function makeMutable<T>(
       initialValue: T
     ): SharedValue<T>;
 
@@ -469,26 +476,27 @@ declare module 'react-native-reanimated' {
     export function useSharedValue<T>(
       initialValue: T,
       shouldRebuild?: boolean
-    ): T extends SharedValueType ? SharedValue<T> : never;
+    ): SharedValue<T>;
 
-    export function useDerivedValue<T extends SharedValueType>(
+    export function useDerivedValue<T>(
       processor: () => T,
       deps?: DependencyList
-    ): SharedValue<T>;
+    ): DerivedValue<T>;
 
     export function useAnimatedReaction<D>(
       dependencies: () => D,
       effects: (dependencies: D) => void,
       deps?: DependencyList
-    );
+    ): void;
                         
     export type AnimatedStyleProp<T extends object> = AnimateStyle<T> | RegisteredStyle<AnimateStyle<T>>;
     export function useAnimatedStyle<
       T extends AnimatedStyleProp<ViewStyle | ImageStyle | TextStyle>
-    >(updater: () => T, deps?: DependencyList): T;
+    >(updater: () => T, deps?: DependencyList | null): T;
     export function useAnimatedProps<T extends {}>(
       updater: () => T,
-      deps?: DependencyList
+      deps?: DependencyList | null,
+      adapters?: PropsAdapterFunction | PropsAdapterFunction[] | null
     ): T;
     export function useAnimatedGestureHandler<
       T extends GestureHandlerGestureEvent = PanGestureHandlerGestureEvent,
@@ -511,6 +519,7 @@ declare module 'react-native-reanimated' {
     ): (...args: Parameters<typeof fn>) => R;
 
     export function useAnimatedRef<T extends Component>(): RefObject<T>;
+    export function defineAnimation<T>(starting: any, factory: () => T): number;
     export function measure<T extends Component>(
       ref: RefObject<T>
     ): {
@@ -730,6 +739,7 @@ declare module 'react-native-reanimated' {
   export const SpringUtils: typeof Animated.SpringUtils;
   export const runOnUI: typeof Animated.runOnUI;
   export const runOnJS: typeof Animated.runOnJS;
+  export const createAnimatedPropAdapter: typeof Animated.createAnimatedPropAdapter;
   export const processColor: typeof Animated.processColor;
   export const makeMutable: typeof Animated.makeMutable;
   export const useValue: typeof Animated.useValue;
@@ -744,6 +754,7 @@ declare module 'react-native-reanimated' {
   export const useAnimatedGestureHandler: typeof Animated.useAnimatedGestureHandler;
   export const useAnimatedScrollHandler: typeof Animated.useAnimatedScrollHandler;
   export const useAnimatedRef: typeof Animated.useAnimatedRef;
+  export const defineAnimation: typeof Animated.defineAnimation;
   export const measure: typeof Animated.measure;
   export const scrollTo: typeof Animated.scrollTo;
   export const withTiming: typeof Animated.withTiming;