diff --git a/node_modules/react-native-reanimated/react-native-reanimated.d.ts b/node_modules/react-native-reanimated/react-native-reanimated.d.ts
index 70ebafa..f3e79d8 100644
--- a/node_modules/react-native-reanimated/react-native-reanimated.d.ts
+++ b/node_modules/react-native-reanimated/react-native-reanimated.d.ts
@@ -186,7 +186,8 @@ declare module 'react-native-reanimated' {
         : S[K] extends object
           ? AnimateStyle<S[K]>
           :
-              | S[K]
+              // allow `number` where `string` normally is to support colors    
+                S[K] extends (string | undefined) ? S[K] | number : S[K]
               | AnimatedNode<
                   // allow `number` where `string` normally is to support colors
                   S[K] extends (string | undefined) ? S[K] | number : S[K]
@@ -387,16 +388,16 @@ declare module 'react-native-reanimated' {
     export function withTiming(
       toValue: number,
       userConfig?: WithTimingConfig,
-      callback?: (isCancelled: boolean) => void,
+      callback?: (isFinished: boolean) => void,
     ): number;
     export function withSpring(
       toValue: number,
       userConfig?: WithSpringConfig,
-      callback?: (isCancelled: boolean) => void,
+      callback?: (isFinished: boolean) => void,
     ): number;
     export function withDecay(
       userConfig: WithDecayConfig,
-      callback?: (isCancelled: boolean) => void
+      callback?: (isFinished: boolean) => void
     ): number;
     export function cancelAnimation<T extends SharedValue<SharedValueType>>(
         sharedValue: T
@@ -434,6 +435,12 @@ declare module 'react-native-reanimated' {
       processor: () => T
     ): SharedValue<T>;
 
+    export function useAnimatedReaction<D>(
+      dependencies: () => D,
+      effects: (dependencies: D) => void,
+      deps?: DependencyList
+    );
+
     export function useAnimatedStyle<T extends StyleProp<AnimateStyle<ViewStyle | ImageStyle | TextStyle>>>(
       updater: () => T
     ): T;
@@ -644,6 +651,7 @@ declare module 'react-native-reanimated' {
   export const useSharedValue: typeof Animated.useSharedValue
   export const useAnimatedStyle: typeof Animated.useAnimatedStyle
   export const useAnimatedProps: typeof Animated.useAnimatedProps
+  export const useAnimatedReaction: typeof Animated.useAnimatedReaction;
   export const useDerivedValue: typeof Animated.useDerivedValue
   export const useAnimatedGestureHandler: typeof Animated.useAnimatedGestureHandler
   export const useAnimatedScrollHandler: typeof Animated.useAnimatedScrollHandler
